lib/src/extend/functions.dart

```diff
@@ -522,15 +522,19 @@ List<List<T>> paths<T>(Iterable<List<T>> choices) => choices.fold(
 QueueList<List<ComplexSelectorComponent>> _groupSelectors(
     Iterable<ComplexSelectorComponent> complex) {
   var groups = QueueList<List<ComplexSelectorComponent>>();
-  var iterator = complex.iterator..moveNext();
-  while (iterator.current != null) {
-    var group = <ComplexSelectorComponent>[];
-    do {
-      group.add(iterator.current);
-    } while (iterator.moveNext() &&
-        (iterator.current is Combinator || group.last is Combinator));
+  var iterator = complex.iterator;
+  if (iterator.moveNext()) {
```

Comment:
```suggestion
  if (!iterator.moveNext()) return groups;
```

Might as well save some indentation.

Reply:
Done.